scipy/stats/_bootstrap.py

```diff
@@ -0,0 +1,393 @@
+    ...    ci = bootstrap(data, np.std, confidence_level=0.9, n_resamples=1000,
+    ...                   random_state=rng).confidence_interval
+    ...    if ci[0] < std_true < ci[1]:
+    ...        ci_contains_true_std += 1
+    >>> print(ci_contains_true_std)
+    873
+
+    Rather than writing a loop, we can also determine the confidence intervals
+    for all 1000 samples at once.
+
+    >>> data = (dist.rvs(size=(n_trials, 100), random_state=rng),)
+    >>> res = bootstrap(data, np.std, axis=-1, confidence_level=0.9,
+    ...                 n_resamples=1000, random_state=rng)
+    >>> ci_l, ci_u = res.confidence_interval
+
+    Here, `ci_l` and `ci_u` contain the confidence interval for each of the
+    ``n_trials = 1000`` samples.
+
+    >>> print(ci_l[995:])
+    [3.7168901  3.69223008 3.41495018 3.3701263  3.45061973]
+    >>> print(ci_u[995:])
+    [4.78001236 4.79348079 4.23119694 4.32124204 4.52526014]
+
+    And again, approximately 90% contain the true value, ``std_true = 4``.
+
+    >>> print(np.sum((ci_l < std_true) & (std_true < ci_u)))
+    897
+
+    `bootstrap` can also be used to estimate confidence intervals of
+    multi-sample statistics, including those calculated by hypothesis
+    tests. `scipy.stats.mood` perform's Mood's test for equal scale parameters,
+    and it returns two outputs: a statistic, and a p-value. To get a
+    confidence interval for the test statistic, we first wrap
+    `scipy.stats.mood` in a function that accepts two sample arguments,
+    accepts an `axis` keyword argument, and returns only the statistic.
+
+    >>> from scipy.stats import mood
+    >>> def my_statistic(sample1, sample2, axis):
+    ...     statistic, _ = mood(sample1, sample2, axis=-1)
+    ...     return statistic
+
+    Here, we use the 'percentile' method with the default 95% confidence level.
+
+    >>> sample1 = norm.rvs(scale=1, size=100, random_state=rng)
+    >>> sample2 = norm.rvs(scale=2, size=100, random_state=rng)
+    >>> data = (sample1, sample2)
+    >>> res = bootstrap(data, my_statistic, method='basic', random_state=rng)
+    >>> print(mood(sample1, sample2)[0])  # element 0 is the statistic
+    -5.521109549096542
+    >>> print(res.confidence_interval)
+    ConfidenceInterval(low=-7.255994487314675, high=-4.016202624747605)
+
+    The bootstrap estimate of the standard error is also available.
+
+    >>> print(res.standard_error)
+    0.8344963846318795
+
+    Paired-sample statistics work, too. For example, consider the Pearson
+    correlation coefficient.
+
+    >>> from scipy.stats import pearsonr
+    >>> n = 100
+    >>> x = np.linspace(0, 10, n)
+    >>> y = x + rng.uniform(size=n)
+    >>> print(pearsonr(x, y)[0])  # element 0 is the statistic
+    0.9962357936065914
+
+    To ensure that samples remain paired, we define a function that accepts
+    an array of _indices_ of the observations for which the statistic is to
+    be calculated.
+
+    >>> def my_statistic(i):
```

Comment:
Rather than using a closure and indices, could this be implemented by creating a new array that is `(n,2)`  (or would it be `(2,n)`?) where each column (or row) would contain a single series?

Reply:
Looks like this wouldn't work.  What about if the data was `(1,n,2)`?

Reply:
> Looks like this wouldn't work.

Do you mean that the example in the docstring will fail? I think it runs successfully.

If the user's data has different dimensions, the user might need to adapt the approach. A feature I suggested at the top was the addition of a paired argument that does this automatically for the user.